### DISCUSSION TITLE
相同元素点不到

### DISCUSSION THREAD

**Discussion by Weslie1999** [gkd-kit/gkd#786] [Announcements]
### 新功能描述

当同一界面有两个相同的元素，希望规则能选择第几个

**Comment by lisonge** [comment]
- 需要给出快照才能知道你到底要干啥

**Reply by Weslie1999** [reply]
我想点击这个第二个button打卡
https://i.gkd.li/snapshot/1731895822251

**Reply by lisonge** [reply]
你从地址栏复制的链接只能你自己看到，别人看不到，要从右上角按钮复制
![image](https://github.com/user-attachments/assets/6731bd73-3e69-49f2-8bfd-6f6b14d5f8ab)

**Reply by lisonge** [reply]
你的选择器是什么，你要选择哪个节点后得到两个结果？可以一次多发点有用的信息，省得我一个一个问

**Reply by Weslie1999** [reply]
https://i.gkd.li/i/17871281
Button[text="打卡"]
选择器选出来的元素有两个，父节点这些都一样，没有id

**Reply by Weslie1999** [reply]
https://i.gkd.li/i/17871483

**Reply by lisonge** [reply]
![image](https://github.com/user-attachments/assets/6d328ffe-7b98-4b4f-a1ec-5d7cbbdfce2e)

```text
Button[text="打卡"][height>50]
```

**Reply by Weslie1999** [reply]
貌似每次，只能切后台一次才能点击到，用这个链接测试就没办法选中
https://i.gkd.li/i/17877007

**Reply by Weslie1999** [reply]
大佬，有解决的办法吗？https://i.gkd.li/i/17877007这个链接用Button[text="打卡"][height>50]
无法选中

**Reply by lisonge** [reply]
你这个应用输出了错乱的无障碍信息，所以高度会异常

选择器的本质就是用自身或周围节点的特征去查询，可以通过对祖先节点进行约束来查询

```text
View[height>100] >3 Button[text="打卡"]
```

**Reply by Weslie1999** [reply]
谢谢，这个关系查询我就是不咋会写，都是从其他应用规则依葫芦画瓢，但又选不中

**Reply by lisonge** [reply]
好的，后续会逐渐完善文档，如果你有关于增加文档的想法也可以发起新的 discussions
